### PR TITLE
RDISCROWD-4830: Python 3.9 fix reserve task category tests

### PR DESCRIPTION
**Pull request by dchhabda**
*Issue number of the reported bug or feature request: #<number>*

**Describe your changes**
A clear and concise description of the changes you have made.

**Testing performed**
Describe the testing you have performed to ensure that the bug has been addressed, or that the new feature works as planned.

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

File: pybossa/redis_lock.py [hunk 1]
```diff
@@ -232,13 +232,17 @@ def get_task_category_lock(self, project_id, user_id=None, category=None, exclud
...
+        category_keys_str = [key.decode() for key in category_keys]
+
         # if key present but for different user, with redundancy = 1, return false
         # TODO: for redundancy > 1, check if additional task run
         # available for this user and if so, return category_key else ""
         if exclude_user:
             # exclude user_id from list of keys passed
             drop_user = ":user:{}:task:".format(user_id)
-            category_keys = [ key for key in category_keys if drop_user not in key ]
+            category_keys = [ key for key in category_keys_str if drop_user not in key ]
```
Comment (XiChenn) [inline pybossa/redis_lock.py:245]: I think you could save a loop by removing line 237 and change these two lines to
```
drop_user = b":user:{}:task:".format(user_id)
category_keys = [ key.decode() for key in category_keys if drop_user not in key ]
```

File: pybossa/redis_lock.py [hunk 2]
```diff
@@ -256,5 +260,6 @@ def acquire_reserve_task_lock(self, project_id, task_id, user_id, category):
 
 
     def release_reserve_task_lock(self, resource_id, pipeline, expiry):
-        cache = pipeline or self._redis
+        #cache = pipeline or self._redis # https://pythonrepo.com/repo/andymccurdy-redis-py-python-connecting-and-operating-databases#locks
+        cache = self._redis
```
Comment (XiChenn) [inline pybossa/redis_lock.py:264]: Are we not using pipeline here? Can we also remove it from the arguments?

File: pybossa/redis_lock.py (hunk 1 above)
Comment (dchhabda) [reply pybossa/redis_lock.py]: This is to return decoded keys to the caller.

File: pybossa/redis_lock.py (hunk 2 above)
Comment (dchhabda) [reply pybossa/redis_lock.py:264]: good point. thanks!

**Review by XiChenn — APPROVED** [review]
lgtm

**Pull request merged**